verisignmdns/src/main/java/denominator/verisignmdns/VerisignMDNSResourceRecordSetApi.java

```diff
@@ -0,0 +1,156 @@
+package denominator.verisignmdns;
+
+import static denominator.common.Preconditions.checkNotNull;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedSet;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import denominator.Credentials;
+import denominator.ResourceRecordSetApi;
+import denominator.common.Util;
+import denominator.model.ResourceRecordSet;
+import denominator.model.Zone;
+import denominator.verisignmdns.VrsnMdns.Record;
+
+public final class VerisignMDNSResourceRecordSetApi implements ResourceRecordSetApi {
+    private final String domainName;
+    private final VrsnMdns api;
+    private final Provider<Credentials> credentialsProvider;
+
+    VerisignMDNSResourceRecordSetApi(Provider<Credentials> credentialsProvider, String domainName, VrsnMdns api) {
+        this.domainName = domainName;
+        this.api = api;
+        this.credentialsProvider = credentialsProvider;
+    }
+
+    @Override
+    public Iterator<ResourceRecordSet<?>> iterator() {
+
+        List<Record> recordList = api.getResourceRecordsList(domainName);
+        return VrsnContentConversionHelper.getSortedSetForDenominator(recordList).iterator();
+    }
+
+    public Iterator<ResourceRecordSet<?>> iterateByNameAndType(String name, String type) {
+        checkNotNull(type, "type was null");
+        checkNotNull(name, "name was null");
+
+        List<Record> recordList = api.getResourceRecordsListForTypeAndName(domainName, type, name);
+        Iterator<ResourceRecordSet<?>> result = VrsnContentConversionHelper.getSortedSetForDenominator(recordList)
+                .iterator();
+        return result;
+    }
+
+    @Override
+    public Iterator<ResourceRecordSet<?>> iterateByName(String name) {
+        try {
+            throw new VrsnMdnsException("Method Not Implemented", -1);
+
+        } catch (Exception ex) {
+            ex.printStackTrace();
+            throw new RuntimeException(ex);
+        }
+    }
+
+    @Override
+    public ResourceRecordSet<?> getByNameAndType(String name, String type) {
+        checkNotNull(type, "type was null");
+        checkNotNull(name, "name was null");
+
+        ResourceRecordSet<?> result = null;
+        List<Record> recordList = api.getResourceRecordsListForTypeAndName(domainName, type, name);
+        SortedSet<ResourceRecordSet<?>> tempSet = VrsnContentConversionHelper.getSortedSetForDenominator(recordList);
+        if (tempSet != null && tempSet.size() > 0) {
+            result = tempSet.first();
+        }
+        return result;
+    }
+
+    @Override
+    public void put(ResourceRecordSet<?> rrset) {
+
+        checkNotNull(rrset, "Resource Record was null");
+        checkNotNull(rrset.name(), "Resource Record Name was null");
+        checkNotNull(rrset.type(), "Resource Record Type was null");
+
+        // At this point disable delete
```

Comment:
put implies deleting records for the same name and type. essentially, the rrset holds the state the user wants: only the records in the set should remain.